src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.vb

```diff
@@ -0,0 +1,435 @@
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub SimpleArrayCreation_UserDefinedType()
+            Dim source = <![CDATA[
+Class M
+End Class
+
+Class C
+    Public Sub F()
+        Dim a = New M() {}'BIND:"New M() {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M())
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub SimpleArrayCreation_ConstantDimension()
+            Dim source = <![CDATA[
+Class M
+End Class
+
+Class C
+    Public Sub F()
+        Const dimension As Integer = 1
+        Dim a = New M(dimension) {}'BIND:"New M(dimension) {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M())
+  IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32, Constant: 2)
+    Left: ILocalReferenceExpression: dimension (OperationKind.LocalReferenceExpression, Type: System.Int32, Constant: 1)
+    Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub SimpleArrayCreation_NonConstantDimension()
+            Dim source = <![CDATA[
+Class M
+End Class
+
+Class C
+    Public Sub F(dimension As Integer)
+        Dim a = New M(dimension) {}'BIND:"New M(dimension) {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M())
+  IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+    Left: IParameterReferenceExpression: dimension (OperationKind.ParameterReferenceExpression, Type: System.Int32)
+    Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
```

Comment:
>IArrayInitializer (OperationKind.ArrayInitializer) [](start = 2, length = 50)

It feels like there should be no initializer for a case like this. I see this as a case where presence of ```{}``` is just a syntax requirement, in fact it is an error to have it non-empty.